models/issue.go

```diff
@@ -1399,3 +1399,41 @@ func updateIssue(e Engine, issue *Issue) error {
 func UpdateIssue(issue *Issue) error {
 	return updateIssue(x, issue)
 }
+
+// IssueList defines a list of issues
+type IssueList []*Issue
+
+func (issues IssueList) loadRepositories(e Engine) ([]*Repository, error) {
+	if len(issues) == 0 {
+		return nil, nil
+	}
+
+	// Load issues.
+	repoIDs := make([]int64, 0, len(issues))
+	for _, issue := range issues {
+		repoIDs = append(repoIDs, issue.RepoID)
+	}
+
+	repositories := make([]*Repository, 0, len(repoIDs))
```

Comment:
It's not the same. We also make `repoMaps := make(map[int64]*Repository, len(repoIDs))` for set repo back to `issue.Repo`.